### ISSUE TITLE
globalVar / routeEnhancers

### ISSUE THREAD

**Issue by prdt3e** [georgringer/news#848]
When I use routeEnhancers, I no longer have access to:
[globalVar = GP: tx_news_pi1|news > 0]
and must use:
[request.getQueryParams () ['tx_news_pi1'] ['news']> 0]

If I use Insertdata to generate a variable(TS) with GP: tx_news_pi1|news, it will not be null.

Thanks

**Comment by georgringer** [comment]
And what should I change?

**Comment by prdt3e** [comment]
If I knew that, I would suggest it to you :)
Possibly is also a core problem?
Or handover routeEnhancers to News.

**Comment by georgringer** [comment]
nothing I can do within EXT:news, I am closing the issue

**Comment by hhchrizzo** [comment]
See https://forge.typo3.org/issues/88252

**Comment by fnagel** [comment]
Trying to get this to work again:

`[globalVar = GP:tx_news_pi1|news > 0] OR [globalVar = GP:tx_news_pi1|news_preview > 0]`

Tried something like this without any success. 

```
[request.getPageArguments().get('tx_news_pi1').get('news') > 0] OR [request.getPageArguments().get('tx_news_pi1').get('news_preview') > 0]

[request.getPageArguments().get('tx_news_pi1|news') > 0] OR [request.getPageArguments().get('tx_news_pi1|news_preview') > 0]

[request.getPageArguments().get('tx_news_pi1.news') > 0] OR [request.getPageArguments().get('tx_news_pi1.news_preview') > 0]

[request.getPageArguments().get('tx_news_pi1[news]') > 0] OR [request.getPageArguments().get('tx_news_pi1[news_preview]') > 0]

[request.getPageArguments().get("tx_news_pi1['news']") > 0] OR [request.getPageArguments().get("tx_news_pi1['news_preview']") > 0]
```

Somebody able to help?

**Comment by mtness** [comment]
since this issue was closed already, stackoverflow might be a better place to get an answer?

**Comment by fnagel** [comment]
Google brings me here, so no.

For anybody interested, take a look here: https://typo3.slack.com/archives/C025BQLFA/p1571309329468900

Spoiler: it's not possible out of the box.

**Comment by liayn** [comment]
see also https://forge.typo3.org/issues/89176

**Comment by xerc** [comment]
please update @ https://docs.typo3.org/p/georgringer/news/master/en-us/AdministratorManual/BestPractice/IntegrationWithTypoScript/Index.html#list-and-detail-on-the-same-page-using-typoscript

**Comment by liayn** [comment]
@georgringer @xerc See https://github.com/georgringer/news/pull/1216

**Comment by w010** [comment]
Sorry for posting in closed thread. I was looking for a solution and came here. I didn't find it in this thread, but I figured it out so I'm posting a solution. Maybe somebody still needs this or will land here like me.

This is the right syntax how to get array values using getPageArguments:
`[request.getPageArguments().get('tx_news_pi1')['news'] > 0]`
It works for me, at least in 10.4.

**Comment by liayn** [comment]
@w010 Or use `[traverse(request.getQueryParams(), 'tx_news_pi1/news') > 0]`

**Comment by xerc** [comment]
or shorter `[request.getQueryParams()['tx_news_pi1']['news']]`

**Comment by w010** [comment]
Yes but then I needed it using explicitly getPageArguments for some reasons. 
> `[request.getPageArguments().get('tx_news_pi1')['news'] > 0]`
> It works for me, at least in 10.4.

[Update, just for archiving] Actually, It turned that it does not, It spams error log when that argument is not specified.
But now nevermind, I used the [traverse(re.... - way, and it seems to work ok in all my cases, and to be the most sure way from all I tested. Only it's name/form is kinda misleading (looks more like some special-use thing, not the most default get/post eval). As of, what I believe it is, direct succesor/migration from good ol' globalVar=GP, which just works, it looks let's say maybe weird for some.
